curator-framework/src/main/java/org/apache/curator/framework/imps/CuratorFrameworkImpl.java

```diff
@@ -257,8 +258,7 @@ public void stateChanged(CuratorFramework client, ConnectionState newState)
 
             client.start();
 
-            executorService = Executors.newFixedThreadPool(2, threadFactory);  // 1 for listeners, 1 for background ops
-
+            executorService = Executors.newSingleThreadScheduledExecutor(threadFactory);
```

Comment:
Is there a reason that this has gone from a 2 thread pool to a single thread pool? Looking at the old code, it would seem that only 1 is used (for the background loop), but just double checking.